portfolio/src/main/webapp/scripts/generateCircles.js

```diff
@@ -0,0 +1,80 @@
+const GOOGLE_COLORS = ["#f4c20d", "#4885ed", "#3cba54"] // yellow, blue, green
+const ICONS = ["contact", "code", "education", "me", "work", "achievement"];
+
```

Comment:
This class gets the job done, for now, but it is not very flexible to future use cases.  You could make it more flexible, and also easier to follow and understand, by reorganizing some of your code into classes.

For instance you could have one class that creates the actual icon element itself, and another that does the math to position it within the parent div.  Something like

const iconGenerator = new IconGenerator(width, height);
const icons = [];
for (const iconName of ["icon1", "icon2", "icon3"]) {
&nbsp;&nbsp;icons.push(IconGenerator.generate(iconName));
}
const iconInjector = new IconInjector(parentDiv, icons.length);
for (const [index, icon] of icons.entries()) {
&nbsp;&nbsp;iconInjector.inject(icon, index)
}

Then in addition to better organized code you will be able to arbitrarily add/remove icons to your heart's content.

Reply:
I thought a lot about this, and did the refactoring for IconGenerate class as you suggested, but for the IconInjector, I'm not too confident to break down the calculations into a separate class since I feel like it belongs to this file.

Could you please take a look at the changes that I made so that we can discuss more about the whole flexibility situation? I have a gut feeling that this would be a premature optimization but probably need more explanation on your side.